src/workerd/io/worker.c++

```diff
@@ -1487,6 +1488,7 @@ Worker::Isolate::~Isolate() noexcept(false) {
 
   metrics->evicted();
   weakIsolateRef->invalidate();
+  *cpuLimitNearlyExceededCallback.lockExclusive() = kj::none;
```

Comment:
Is there a lifetime issue here which requires this to be destroyed before the rest of the Isolate? I guess it'd just be weird if the function we pass to `setCpuLimitNearlyExceededCallback()` were to outlive parts of the Isolate?